contracts/src/token/erc721/extensions/wrapper.rs

```diff
@@ -0,0 +1,85 @@
+//! Extension of the ERC-721 token contract to support token wrapping.
+//!
+//! Users can deposit and withdraw an "underlying token" and receive a "wrapped
+//! token" with a matching tokenId. This is useful in conjunction with other
+//! modules.
+use alloc::{vec, vec::Vec};
+
+use alloy_primitives::{Address, U256};
+use stylus_sdk::{contract, prelude::storage, stylus_proc::SolidityError};
+
+use crate::token::{
+    erc721,
+    erc721::{ERC721IncorrectOwner, Erc721, IErc721},
+};
+
+/// State of an [`Erc721Wrapper`] token.
+#[storage]
+pub struct Erc721Wrapper {
+    /// Erc721 contract storage.
+    pub _underlying: Erc721,
```

Comment:
@bidzyyys What should the type of `_underlying` be?

Reply:
You should probably be using `pub _underlying: Address`, as this is an external contract, and Stylus currently has no way to represent them as part of structs.

You'll then also probably need to define a local sol interface, like in [VestingWallet](https://github.com/OpenZeppelin/rust-contracts-stylus/blob/3205fc7a9bff59af753d4fef9253343ae2c631b8/contracts/src/finance/vesting_wallet.rs#L92-L102).